### PR TITLE
Kafka example

### PR DESCRIPTION
**Pull request by zbendhiba**
Fixes #1028

Note that the `main` branch points at the latest stable Camel Quarkus release.
Pull requests should be generally send against the `camel-quarkus-main` branch pointing at the current Camel Quarkus SNAPSHOT.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kafka/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## ---------------------------------------------------------------------------
+
+#Kafka topic Name
+kafka.topic.name=test
+
+# How often should the books CSV be generated
```
Comment (jamesnetherton) [inline kafka/src/main/resources/application.properties:21]: I guess this comment needs updating.

**Review by ppalaga — COMMENTED** [review]
Wow, nice work!

Could you please run `./mvnw org.l2x6.cq:cq-maven-plugin:0.33.0:update-examples-json` from the  repo root dir to update the json file used by antora to produce this page https://camel.apache.org/camel-quarkus/latest/user-guide/examples.html ?

If we really need all five Dockerfiles, it would be nice to add a sentence explaining how/when to use which (maybe it is documented somewhere, and I just oversaw it?). Or maybe link the Quarkus guide explaining it. But perhaps we do not need all five?

**Review by jamesnetherton — APPROVED** [review]
Please add the module into https://github.com/apache/camel-quarkus-examples/blob/main/.github/test-categories.yaml.

File: kafka/src/main/resources/application.properties (hunk 1 above)
Comment (zbendhiba) [reply kafka/src/main/resources/application.properties:21]: yes !!

**Comment by zbendhiba** [comment]
> If we really need all five Dockerfiles, it would be nice to add a sentence explaining how/when to use which (maybe it is documented somewhere, and I just oversaw it?). Or maybe link the Quarkus guide explaining it. But perhaps we do not need all five?

The fast-jar obviously not generated anymore on code.quarkus.
I don't find explanation, but I think we don't need more than 2. I guess the legacy-jar is for apps before 1.12.x, before the move to fast-jar. I don't know what distroless container is;
I'll leave just jvm and native

**Comment by zbendhiba** [comment]
I forgot to add the license.

**Comment by zbendhiba** [comment]
I don't know what needs to be done.
- updated the json example file
- checked the license and format
- add a new category (please check if okay or if I need to come up with a better name ?)
I'm trying to see what's missing from the ci-build.yaml and tested the mvnw-for-each.sh, can't have those uncommited changes on my machine.

Do we need to merge main in camel-quarkus-main or something ?

**Comment by jamesnetherton** [comment]
I think you'll need to run `./mvnw-for-each.sh org.l2x6.cq:cq-maven-plugin:0.33.0:sync-example-properties` then commit & push the resulting changes to the PR.

**Comment by zbendhiba** [comment]
@jamesnetherton thanks.

**Pull request merged**